src/base/bittorrent/endpointview.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2020  Kacper Michajłow <kasper93@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#pragma once
+
+#include <libtorrent/fwd.hpp>
+
+class QString;
+
+namespace BitTorrent
+{
+    enum class TrackerStatus;
+
+    class EndpointView
+    {
+    public:
+        EndpointView(const lt::announce_endpoint &nativeEndpoint);
+
+        QString name() const;
+        int numSeeds() const;
+        int numLeeches() const;
+        int numDownloaded() const;
+        QString message() const;
+        TrackerStatus status() const;
+
+    private:
+        const lt::announce_endpoint &m_nativeEndpoint;
```

Comment:
I'm not sure it's safe to store reference here.

Reply:
It is. This class is not intended to store any state or be stored anywhere. It is intended only as a temporary view for announce entry.

Reply:
> This class is not intended to store any state or be stored anywhere.

But you do store it in container and make it accessible via public interface of TrackerEntry.

Reply:
But as long as TrackerEntry lives those will live too. And you need to think how would anyone use this. Copy or not  `lt::announce_endpoint` you would keep, you want to peek into it and throw away. If you copy, state would not even change so what would be the point of storing it? I mean you can't simply say storing reference in class is __always__ unsafe, because everything have assumptions and restrictions.

Reply:
>If you copy, state would not even change so what would be the point of storing it? 

There may be something not so obvious here... For example, get list of endpoints from a temporary and then process it.
> I mean you can't simply say storing reference in class is always unsafe

Yes, I can't. And I don't. Here is one of the special cases. It's unclear from TrackerEntry interface that there's list of some temporaries.